src/resources/utils/arrayUtils.tsx

```diff
@@ -8,3 +13,39 @@ export function arraysEqual<T>(array1?: T[], array2?: T[]): boolean {
   }
   return true;
 }
+
+/**
+ * Paginates an array, returning the array elements pertaining to the given page.
+ * @param array The array to be paginated.
+ * @param page The page number of the returned elements. (Indexation starts at 1)
+ * @param numPerPage Number of elements on each page.
+ * @returns The elements present on the given page.
+ * @throws {RangeError} Thrown when either numPerPage or page is not valid (e.g., asking for a page that does not exist).
+ */
+export function getArrayPage<T>(array: T[], page: number, numPerPage: number): T[] {
+  if (numPerPage <= 0) {
+    throw new RangeError('numPerPage must be greater than 0');
+  }
+
+  if (page <= 0 || page > Math.ceil(array.length / numPerPage)) {
+    throw new RangeError('Invalid page number');
+  }
+
+  const pageStartIndex = (page - 1) * numPerPage;
+  const endIndex = Math.min(pageStartIndex + numPerPage, array.length);
+
+  return array.slice(pageStartIndex, endIndex);
+}
+
+/**
+ * Calculates the total number of pages needed to paginate an array.
+ * @param array The array to be paginated.
+ * @param numPerPage Number of elements on each page.
+ * @returns The total number of pages.
+ */
+export function getTotalNumOfPages<T>(array: T[], numPerPage: number): number {
+  if (numPerPage <= 0) {
+    throw new RangeError('numPerPage must be greater than 0');
+  }
+  return Math.ceil(array.length / numPerPage);
```

Comment:
It will probably not happen for the user list, but if a paged array has no items, this function will return 0. If the user list has no items, the `getArrayPage` function will throw `new RangeError('Invalid page number');`. Should we return total number of pages === 1 when an array has no items?

Reply:
Good catch! We absolutely should!
Adding that and a test to it now! 👍